src/Features/Core/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -54,7 +54,12 @@ public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this Diag
                         category |= DiagnosticAnalyzerCategory.ProjectAnalysis;
                     }
 
-                    if (!cantSupportSemanticSpanAnalysis)
+                    // Current DiagnosticAnalyzer API has no restriction on any analyzer action reporting diagnostics anywhere within a compilation.
+                    // Hence, we must conservatively assume they can't support incremental span based analysis across compilations.
+                    // GitHub issue https://github.com/dotnet/roslyn/issues/1565 tracks adding support for marking an analyzer incremental.
+                    // Until then, we just assume that all public/non-built in analyzers can never support incremental analysis.
+
+                    if ((analyzer is IBuiltInAnalyzer) && !cantSupportSemanticSpanAnalysis)
```

Comment:
I am not sure how unused usings can support incremental method body analysis, it needs to be a full compilation unit/tree level diagnostic. Any edit in source file will need recomputing unused usings right?

I don't get the context of this comment: "I think you should remove it for solution crawler path, but not for LB path". 
Did you mean for public analyzers? If so, then how can we use incremental span analysis even for LB path for public analyzers if we are not sure if they are incremental or not?
Did you mean for built-in analyzers? If so, what is the new check you propose over what we currently have within this if statement?